src/wp-includes/interactivity-api/class-wp-interactivity-api.php

```diff
@@ -0,0 +1,672 @@
+	 * that it overrides the previous ones.
+	 *
+	 * @since 6.5.0
+	 *
+	 * Example:
+	 *
+	 *     set_style_property( 'color:green;', 'color', 'red' )      => 'color:red;'
+	 *     set_style_property( 'background:green;', 'color', 'red' ) => 'background:green;color:red;'
+	 *     set_style_property( 'color:green;', 'color', null )       => ''
+	 *
+	 * @param string            $style_attribute_value The current style attribute value.
+	 * @param string            $style_property_name   The style property name to set.
+	 * @param string|false|null $style_property_value  The value to set for the style property. With false, null or an
+	 *                                                 empty string, it removes the style property.
+	 * @return string The new style attribute value after the specified property has been added, updated or removed.
+	 */
+	private function set_style_property( string $style_attribute_value, string $style_property_name, $style_property_value ): string {
+		$style_assignments    = explode( ';', $style_attribute_value );
+		$result               = array();
+		$style_property_value = ! empty( $style_property_value ) ? rtrim( trim( $style_property_value ), ';' ) : null;
+		$new_style_property   = $style_property_value ? $style_property_name . ':' . $style_property_value . ';' : '';
+
+		// Generate an array with all the properties but the modified one.
+		foreach ( $style_assignments as $style_assignment ) {
+			if ( empty( trim( $style_assignment ) ) ) {
+				continue;
+			}
+			list( $name, $value ) = explode( ':', $style_assignment );
+			if ( trim( $name ) !== $style_property_name ) {
+				$result[] = trim( $name ) . ':' . trim( $value ) . ';';
+			}
+		}
+
+		// Add the new/modified property at the end of the list.
+		array_push( $result, $new_style_property );
+
+		return implode( '', $result );
+	}
+
+	/**
+	 * Processes the `data-wp-text` directive.
+	 *
+	 * It updates the inner content of the current HTML element based on the
+	 * evaluation of its associated reference.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param WP_Interactivity_API_Directives_Processor $p               The directives processor instance.
+	 * @param array                                     $context_stack   The reference to the context stack.
+	 * @param array                                     $namespace_stack The reference to the store namespace stack.
+	 */
+	private function data_wp_text_processor( WP_Interactivity_API_Directives_Processor $p, array &$context_stack, array &$namespace_stack ) {
+		if ( ! $p->is_tag_closer() ) {
+			$attribute_value = $p->get_attribute( 'data-wp-text' );
+			$result          = $this->evaluate( $attribute_value, end( $namespace_stack ), end( $context_stack ) );
+
+			/*
+				* Follows the same logic as Preact in the client and only changes the
+				* content if the value is a string or a number. Otherwise, it removes the
+				* content.
+				*/
```

Comment:
```suggestion
			/*
			 * Follows the same logic as Preact in the client and only changes the
			 * content if the value is a string or a number. Otherwise, it removes the
			 * content.
			 */
```